config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/JmsDestinationEntityBuilder.java

```diff
@@ -0,0 +1,283 @@
+        }
+        
+        boolean isTemplate = false;
+
+        if (isInbound) {
+            this.buildInboundDestination(bundle, name, jmsDestination, jmsDestinationDetailProps, contextPropertiesTemplateProps);
+        } else {
+            isTemplate = jmsDestination.getOutboundDetail().isTemplate();
+            this.buildOutboundDestination(jmsDestination, jmsDestinationDetailProps, contextPropertiesTemplateProps);
+        }
+
+        if (jmsDestination.getAdditionalProperties() != null && !jmsDestination.getAdditionalProperties().isEmpty()) {
+            contextPropertiesTemplateProps.putAll(jmsDestination.getAdditionalProperties());
+        }
+        jmsDestinationDetailEle.appendChild(createElementWithTextContent(document, ENABLED, true));
+        jmsDestinationDetailEle.appendChild(createElementWithTextContent(document, TEMPLATE, isTemplate));
+        buildAndAppendPropertiesElement(jmsDestinationDetailProps, document, jmsDestinationDetailEle);
+
+        // Build JMS Connection element.
+        String jmsConnectionEleId = idGenerator.generate();
+        Element jmsConnectionEle = createElementWithAttributesAndChildren(
+                document,
+                JMS_CONNECTION,
+                ImmutableMap.of(ATTRIBUTE_ID, jmsConnectionEleId),
+                createElementWithTextContent(document, TEMPLATE, isTemplate)
+        );
+
+        Map<String, Object> jmsConnectionProps = new HashMap<>();
+        jmsConnectionProps.put(JNDI_INITIAL_CONTEXT_FACTORY_CLASSNAME, jmsDestination.getInitialContextFactoryClassName());
+        jmsConnectionProps.put(JNDI_PROVIDER_URL, jmsDestination.getJndiUrl());
+        jmsConnectionProps.put(CONNECTION_FACTORY_NAME, jmsDestination.getConnectionFactoryName());
+        jmsConnectionProps.put(PROPERTY_USERNAME, jmsDestination.getDestinationUsername());
+        if (jmsDestination.getDestinationPasswordRef() != null) {
+            jmsConnectionProps.put(PROPERTY_PASSWORD, String.format(STORED_PASSWORD_REF_FORMAT, jmsDestination.getDestinationPasswordRef()));
+        } else {
+            jmsConnectionProps.put(PROPERTY_PASSWORD, jmsDestination.getDestinationPassword());
+        }
+        buildAndAppendPropertiesElement(jmsConnectionProps, document, jmsConnectionEle);
+        jmsConnectionEle.appendChild(buildPropertiesElement(contextPropertiesTemplateProps, document, CONTEXT_PROPERTIES_TEMPLATE));
+
+        // Build JMS Destination element.
+        Element jmsDestinationEle = createElementWithAttributesAndChildren(
+                document,
+                JMS_DESTINATION,
+                ImmutableMap.of(ATTRIBUTE_ID, id));
+        jmsDestinationEle.appendChild(jmsDestinationDetailEle);
+        jmsDestinationEle.appendChild(jmsConnectionEle);
+
+        return EntityBuilderHelper.getEntityWithNameMapping(EntityTypes.JMS_DESTINATION_TYPE, name, id, jmsDestinationEle);
+    }
+    
+    private void buildInboundDestination (
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. ([squid:S3776](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS3776))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=138&issues=AWg5aom1Si5GWQS9DSyM&open=AWg5aom1Si5GWQS9DSyM)

[ISSUE_KEY]: AWg5aom1Si5GWQS9DSyM